cmd/aws-actuator/main.go

```diff
@@ -221,210 +202,96 @@ func readMachineManifest(manifestParams *manifestParams, manifestLoc string) (*c
 			}
 
-			glog.Infof("Reading cluster manifest from %v", path.Join(manifestsDir, "cluster.yaml"))
-			cluster, err := readClusterManifest(path.Join(manifestsDir, "cluster.yaml"))
-			if err != nil {
-				return err
+			testNamespace := &apiv1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
 			}
 
-			glog.Infof("Reading master machine manifest from %v", path.Join(manifestsDir, "master-machine.yaml"))
-			masterMachine, err := readMachineManifest(
-				&manifestParams{
-					ClusterID: machinePrefix,
+			testCluster := &clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      machinePrefix,
+					Namespace: testNamespace.Name,
 				},
-				path.Join(manifestsDir, "master-machine.yaml"),
+				Spec: clusterv1.ClusterSpec{
+					ClusterNetwork: clusterv1.ClusterNetworkingConfig{
+						Services: clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.0.0.1/24"},
+						},
+						Pods: clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.0.0.1/24"},
+						},
+						ServiceDomain: "example.com",
+					},
+				},
+			}
+
+			awsCredentialsSecret := testutils.GenerateAwsCredentialsSecretFromEnv(awsCredentialsSecretName, testNamespace.Name)
+
+			// Create master machine and verify the master node is ready
+			masterUserDataSecret, err := manifests.MasterMachineUserDataSecret(
+				"masteruserdatasecret",
+				testNamespace.Name,
+				[]string{"\\$(curl -s http://169.254.169.254/latest/meta-data/public-hostname)", "\\$(curl -s http://169.254.169.254/latest/meta-data/public-ipv4)"},
 			)
 			if err != nil {
 				return err
 			}
 
-			glog.Infof("Reading master user data manifest from %v", path.Join(manifestsDir, "master-userdata.yaml"))
-			masterUserDataSecret, err := readSecretManifest(path.Join(manifestsDir, "master-userdata.yaml"))
+			masterMachineProviderConfig, err := testutils.MasterMachineProviderConfig(awsCredentialsSecret.Name, masterUserDataSecret.Name, testCluster.Name)
 			if err != nil {
 				return err
 			}
 
-			if machinePrefix != "" {
-				masterMachine.Name = machinePrefix + "-" + masterMachine.Name
-			}
-
-			var awsCredentialsSecret *apiv1.Secret
-			if cmd.Flag("aws-credentials").Value.String() != "" {
-				glog.Infof("Reading aws credentials manifest from %v", cmd.Flag("aws-credentials").Value.String())
-				awsCredentialsSecret, err = readSecretManifest(cmd.Flag("aws-credentials").Value.String())
-				if err != nil {
-					return err
-				}
-			}
+			masterMachine := manifests.MasterMachine(testCluster.Name, testCluster.Namespace, masterMachineProviderConfig)
 
 			glog.Infof("Creating master machine")
-			actuator := createActuator(masterMachine, awsCredentialsSecret, masterUserDataSecret)
-			result, err := actuator.CreateMachine(cluster, masterMachine)
+
+			actuator := utils.CreateActuator(masterMachine, awsCredentialsSecret, masterUserDataSecret)
+			result, err := actuator.CreateMachine(testCluster, masterMachine)
+			glog.Error(err)
```

Comment:
Why this unconditionally?

Reply:
Huh, that was supposed to be in the `if err != nil` :-|

Reply:
fixed